planner/core/integration_test.go

```diff
@@ -7060,6 +7060,31 @@ func TestHexIntOrStrPushDownToTiFlash(t *testing.T) {
 	tk.MustQuery("explain select hex(b) from t;").CheckAt([]int{0, 2, 4}, rows)
 }
 
+func TestBinPushDownToTiFlash(t *testing.T) {
```

Comment:
Maybe you can add some unit tests like [this](https://github.com/pingcap/tiflash/blob/master/dbms/src/Common/tests/gtest_mytime.cpp)

Reply:
Do you means some `select` tests can be added there?

I found that nearly all the integration tests of `push down function` like  `TestTimeToSecPushDownToTiFlash`, `TestReverseUTF8PushDownToTiFlash`, `TestHexIntOrStrPushDownToTiFlash` are only check the explain result.

It is because that the related push down functions are checked and tested in tiflash, so do the [`bin`](https://github.com/pingcap/tiflash/blob/master/dbms/src/Functions/tests/gtest_bin.cpp).

Reply:
> Do you means some `select` tests can be added there?
> 
> I found that nearly all the integration tests of `push down function` like `TestTimeToSecPushDownToTiFlash`, `TestReverseUTF8PushDownToTiFlash`, `TestHexIntOrStrPushDownToTiFlash` are only check the explain result.
> 
> It is because that the related push down functions are checked and tested in tiflash, so do the [`bin`](https://github.com/pingcap/tiflash/blob/master/dbms/src/Functions/tests/gtest_bin.cpp).

no, the tests here are integration test, I mean that adding some unit tests, you can refer the unit tests of function reverse  here https://github.com/pingcap/tiflash/blob/c9ce3180879a084db20b1db905b59ab4caab3a3a/dbms/src/Functions/tests/gtest_strings_reverse.cpp#L47

Reply:
Emm... Where should the unit tests be added? Here (integration_test.go)?

Reply:
IMO, the test in `integration_test.go` is mock test, so that all the sql queries must be executed based on the mock db, not TiFlash. So adding more unit tests in TiDB to check the result is unnecessary.

On the other hand, I think there are [enough unit tests](https://github.com/pingcap/tiflash/blob/master/dbms/src/Functions/tests/gtest_bin.cpp) to check the `bin` function is implemented correctly in TiFlash. These unit tests are written nearly like the `reverse` function you mentioned above.

Reply:
> IMO, the test in `integration_test.go` is mock test, so that all the sql queries must be executed based on the mock db, not TiFlash. So adding more unit tests in TiDB to check the result is unnecessary.
> 
> On the other hand, I think there are [enough unit tests](https://github.com/pingcap/tiflash/blob/master/dbms/src/Functions/tests/gtest_bin.cpp) to check the `bin` function is implemented correctly in TiFlash. These unit tests are written nearly like the `reverse` function you mentioned above.

OK.